README.md

```diff
@@ -141,30 +151,72 @@ fn main() -> pgp::errors::Result<()> {
-- [`rpgpie`]: An experimental high level OpenPGP API
-- [`rsop`]: A SOP CLI tool based on rPGP and rpgpie
-- [`debian-packaging`]: a library crate for dealing with Debian packages
+> rPGP allows following almost all parts of the OpenPGP specification, but the APIs are low level building blocks and do not claim that using them is (a) secure or (b) following the OpenPGP specification
 
-Don't see your project here? Please send a PR :)
+Using the building blocks in rPGP correctly and safely requires a solid understanding of OpenPGP and at least a basic understanding of cryptography.
 
-### FAQs
+### OpenPGP is a layered technology
 
-Checkout [FAQ.md](docs/FAQ.md).
+For context, OpenPGP can be thought of as a multi-layered technology, roughly like this:
+
+1. Wire format: [Packet framing](https://www.rfc-editor.org/rfc/rfc9580.html#name-packet-syntax), [Packet content](https://www.rfc-editor.org/rfc/rfc9580.html#name-packet-types), [ASCII armor](https://www.rfc-editor.org/rfc/rfc9580.html#name-forming-ascii-armor), ...
+2. Composite objects (e.g. Certificates, Messages) constructed according to [grammars](https://www.rfc-editor.org/rfc/rfc9580.html#name-packet-sequence-composition)
+3. Functionality to process data, such as calculation and validation of OpenPGP signatures and encryption and decryption of messages.
+4. OpenPGP semantics (e.g.: *Expiration* and *Revocation* of Certificates and their components, *Key Flags* that define which semantical operations a given component key may be used for, signaling of algorithm preferences, ...)
+
+Of these layers, the OpenPGP RFC specifies 1-3, while 4 is not specified in detail.
+
+Like the RFC, rPGP handles layers 1-3, but explicitly does not deal with 4.
+Applications that need OpenPGP semantics must implement them manually, or rely on additional libraries to deal with that layer.
+
+Some work on formalizing OpenPGP semantics can be found in
+[draft-gallagher-openpgp-signatures](https://datatracker.ietf.org/doc/draft-gallagher-openpgp-signatures/) and
+[draft-dkg-openpgp-revocation](https://datatracker.ietf.org/doc/draft-dkg-openpgp-revocation/).
+
+### rPGP doesn't implement support for dealing with "OpenPGP semantics"
+
+As outlined above, rPGP does not deal with the "semantics layer" of OpenPGP
+(such as the *expiration* or *revocation* status of certificates, or their components).
+Applications in which these semantics are relevant must ensure that they implement them separately.
+
+The same applies for abstractions to determine the semantical validity of component keys for a particular operation
+(such as: "which component key, if any, is currently valid for the verification of data signatures?").
+
+NOTE: The [`rpgpie`] library implements some of these high level OpenPGP semantics.
+It may be useful either to incorporate in rPGP projects, or to study for reference.
+
+### Legacy mechanisms are available (but using them may be insecure)
+
```

Comment:
I wonder if this could be folded better into the layering approach, by adding a time dimension and saying that in addition to the layers, there is the time dependent evolution of the spec